images/ad-server/install-sambacc.sh

```diff
@@ -3,18 +3,46 @@
 set -ex
 
 wheeldir="$1"
-container_json_file="$2"
+if ! [ -d "${wheeldir}" ]; then
+    echo "no directory: ${wheeldir}"
+    exit 2
+fi
+
+mapfile -d '' wheels < \
+    <(find "${wheeldir}" -type f -name 'sambacc-*.whl' -print0)
+mapfile -d '' rpmfiles < \
+    <(find "${wheeldir}" -type f -name '*sambacc-*.noarch.rpm' -print0)
 
-wheeldir=/tmp
-wheel="$(find "${wheeldir}" -type f -name 'sambacc-*.whl')" \
 
-if ! [ "$(echo "$wheel" | wc -l)" = 1 ]; then
+if [ "${#wheels[@]}" -gt 1 ]; then
     echo "more than one wheel file found"
     exit 1
+elif [ "${#wheels[@]}" -eq 1 ]; then
+    action=install-wheel
+fi
+
+if [ "${#rpmfiles[@]}" -gt 1 ]; then
+    echo "more than one rpm file found"
+    exit 1
+elif [ "${#rpmfiles[@]}" -eq 1 ]; then
+    action=install-rpm
 fi
```

Comment:
Can you explain the logic here? Why do we insist on having just a single rpm? Under what circumstances we expect more than one?

Reply:
The build container should only generate exactly one RPM (recent versions) and exactly one wheel (traditionally) but we also dont know for sure that the input is the result of the sambacc build container. Ensuring that we only have 1 RPM or wheel in the source dir is a simplistic way to try and double check that our input dir is what we expect.  If you mounted a dir from your host that had multiple extra files in it would be one way to trigger the condition where there are multiple RPMs or wheels.